### DISCUSSION TITLE
i got dotnet48 working on linux and scripthookVdotnet but...

### DISCUSSION THREAD

**Discussion by kowalski713** [scripthookvdotnet/scripthookvdotnet#1414] [Q&A]
Hello first i want to say my technical knowledge is very low but i was able to find a work around to dotnet48 on linux i posted [here](https://lemmy.dbzer0.com/post/17099008) and the game loads with the scripthookdotnet dlls.  just cant seem to load any mods.  Not asking for a fix, i understand im on my own on linux. just looking for some help understanding the logs to help point me in the right direction

this is the scripthookVdotnet log
`
[22:47:00] [DEBUG] Loading API from .\ScriptHookVDotNet2.dll ...
[22:47:00] [DEBUG] Loading API from .\ScriptHookVDotNet3.dll ...
[22:47:00] [DEBUG] Loading scripts from Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts ...
[22:47:00] [DEBUG] Loading assembly AdvancedTurnSignals.dll ...
[22:47:00] [ERROR] Unable to load AdvancedTurnSignals.dll: System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'AdvancedTurnSignals, Version=1.2.3.0, Culture=neutral, PublicKeyToken=null'.
File name: 'AdvancedTurnSignals, Version=1.2.3.0, Culture=neutral, PublicKeyToken=null' ---> System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\AdvancedTurnSignals.dll'.
File name: 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\AdvancedTurnSignals.dll'

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at SHVDN.ScriptDomain.LoadScriptsFromAssembly(String filename)

[22:47:00] [DEBUG] Loading assembly Disarm.dll ...
[22:47:00] [ERROR] Unable to load Disarm.dll: System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'Disarm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
File name: 'Disarm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' ---> System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\Disarm.dll'.
File name: 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\Disarm.dll'

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at SHVDN.ScriptDomain.LoadScriptsFromAssembly(String filename)

[22:47:00] [DEBUG] Loading assembly SmoothDrivingV.dll ...
[22:47:00] [ERROR] Unable to load SmoothDrivingV.dll: System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'SmoothDrivingV, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null'.
File name: 'SmoothDrivingV, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null' ---> System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\SmoothDrivingV.dll'.
File name: 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\SmoothDrivingV.dll'

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at SHVDN.ScriptDomain.LoadScriptsFromAssembly(String filename)

[22:47:00] [DEBUG] Loading assembly Stance.dll ...
[22:47:00] [ERROR] Unable to load Stance.dll: System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'Stance, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
File name: 'Stance, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' ---> System.IO.FileLoadException: The specified user does not have a valid profile.  Unable to load 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\Stance.dll'.
File name: 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\Stance.dll'

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at SHVDN.ScriptDomain.LoadScriptsFromAssembly(String filename)

`

**Comment by Abdulhanan535** [comment]
EVERYONE needs him

**Comment by Abdulhanan535** [comment]
bro will you help me too?

**Reply by kagikn** [reply]
Sorry, but I should address #1454 before I try to fiddle with my Linux environment so I can find a stable solution about SHVDN on Linux. Unfortunately I only have a little knowledge about Wine, too.

**Reply by Abdulhanan535** [reply]
okay, lets see then what i can do

**Reply by Abdulhanan535** [reply]
though wine is just like windows all required libraries that run a game or window application wine is a package for those

**Comment by nomakewan** [comment]
So since the error displayed is `System.IO.FileLoadException: The specified user does not have a valid profile. Unable to load 'file:///Z:\home\kowalski\modded\steamapps\common\Grand Theft Auto V\scripts\Disarm.dll'` on all of these, I wonder if it's a permissions issue on the scripts directory. Which group:user owns that folder? What are the read/write/execute settings on that folder?

Could be as easy as a chown -R and chmod -R. Wouldn't that be cool.

**Comment by MaKraMc** [comment]
It's actually pretty easy to get it running under Linux! :)
I used protontricks (which was a bit fiddly at first) to launch winecfg in the GTA5 prefix (steam version)

From there go to Libraries and add `dinput8` or `dinput8.dll` like so:
![image](https://github.com/user-attachments/assets/a7540213-c249-4fcc-81f6-c27cb65fe250)

Then launch the game with ScriptHook installed an it shows up. Hope this helps!

I figure it would be even easier in a Launcher like Heroic, which has a built-in button to launch winecfg! Happy modding :)

[This article](https://gtamods.com/wiki/Linux#Without_Steam) gave me some hints that wine is loading the built-in library so I figured to just add the override an it worked.

**Reply by MaKraMc** [reply]
No seperate dotnet 48 installation needed

**Reply by DividedGibon** [reply]
This only covers ScriptHookV.  ScripthookVdotnet is a different beast entirely.  I've got a separate GIT issue re getting Steamdeck functioning.  After weeks of trying i've finally got DOT NET Scripthook running.  

The steps are a bit more complicated than anything I've done before to get modding working on Steam Deck.  Particularly as, at least from my testing, Protontricks is not enough and winetricks is necessary. 

I need to retest my method before documenting and sharing. It was full of pitfalls which required workarounds and I don't want to include those false steps.

**Reply by MaKraMc** [reply]
Thanks for pointing this out. I was not aware of this version. Sorry for spreading false hope

**Comment by DividedGibon** [comment]
This only covers ScriptHookV. ScripthookVdotnet is a different beast entirely. I've got a separate GIT issue re getting Steamdeck functioning. After weeks of trying i've finally got DOT NET Scripthook running.

The steps are a bit more complicated than anything I've done before to get modding working on Steam Deck. Particularly as, at least from my testing, Protontricks is not enough and winetricks is necessary.

I need to retest my method before documenting and sharing. It was full of pitfalls which required workarounds and I don't want to include those false steps.

See[ issue here](https://github.com/scripthookvdotnet/scripthookvdotnet/issues/1506) where I'll u[update
https://github.com/scripthookvdotnet/scripthookvdotnet/issues/1506

**Comment by Abdulhanan535** [comment]
lets see. i can finally run modded scripts now.

**Comment by dim-ghub** [comment]
I solved this! proton tricks -> GTAV -> regedit -> delete `HKLM\Software\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-21-0-0-0-1000`

then in a terminal emulator:
```
export WINEPREFIX=~/.local/share/Steam/steamapps/compatdata/271590/pfx/
wine reg add "HKLM\\Software\\Microsoft\\Windows NT\\CurrentVersion\\ProfileList\\S-1-5-21-0-0-0-1000"
```

No clue why this worked for me but it did!

**Reply by nomakewan** [reply]
> I solved this! proton tricks -> GTAV -> regedit -> delete `HKLM\Software\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-21-0-0-0-1000`
> 
> then in a terminal emulator:
> 
> ```
> export WINEPREFIX=/home/win7/.local/share/Steam/steamapps/compatdata/271590/pfx/
> wine reg add "HKLM\\Software\\Microsoft\\Windows NT\\CurrentVersion\\ProfileList\\S-1-5-21-0-0-0-1000"
> ```
> 
> No clue why this worked for me but it did!

Could you please show your copy of GTAV in Linux/Steam Deck actually running a .NET script? Lots of people have claimed to have "fixed" this but no one has actually shown an SHVDN script actually running.

**Reply by dim-ghub** [reply]
Of course!

https://github.com/user-attachments/assets/e395a2e7-af57-45fe-a62d-ee1830082780

**Reply by nomakewan** [reply]
> Of course!
> 2025-03-24.20-45-35.mp4

Impressive not only because it shows running scripts, but because of how long it took me to be convinced that that _isn't_ Windows 7. Very, very impressive theming job, right down to the command prompt header.

Congrats on being the first person to show off SHVDN 3.7.0.20 working under Linux!!

**Reply by dim-ghub** [reply]
Don't forget the PS1 prompt `C:\home\win7`

But, I'm definitely not the first, I've been using shvdnv3 before on Linux, these errors only started occurring today

**Comment by islammuhammad773-max** [comment]
[21:33:04] [DEBUG] Loading API from .\ScriptHookVDotNet2.dll ...
[21:33:05] [DEBUG] Loading API from .\ScriptHookVDotNet3.dll ...
[21:33:05] [DEBUG] Initializing NativeMemory members...
[21:33:18] [DEBUG] Loading scripts from d:\Grand Theft Auto V\scripts ...
[21:33:23] [DEBUG] Successfully compiled Roadrage.cs using API version 3.7.0. If you find the script not working as the author(s) intended, you could annotate an API version by adding a dot and a single-digit number for API version before the extension (e.g. "Roadrage.2.cs").
[21:33:23] [INFO] Found 1 script(s) in Roadrage.cs resolved to API version 3.7.0 (script assembly version: 0.0.0.0, target API version: 3.7.0).
[21:33:23] [DEBUG] Loading assembly ArrestWarrant.dll ...
[21:33:23] [WARNING] Resolving API version 0.0.0 referenced in ArrestWarrant.dll.
[21:33:23] [INFO] Found 1 script(s) in ArrestWarrant.dll resolved to API version 2.11.6 (script assembly version: 1.0.0.0, target API version: 2.8.0).
[21:33:23] [DEBUG] Loading assembly Better Chases.dll ...
[21:33:24] [INFO] Found 2 script(s) in Better Chases.dll resolved to API version 2.11.6 (script assembly version: 1.0.0.0, target API version: 2.9.6).
[21:33:24] [DEBUG] Loading assembly LemonUI.SHVDN3.dll ...
[21:33:24] [INFO] Found 0 script(s) in LemonUI.SHVDN3.dll (script assembly version: 2.1.2.0, target API version: 3.6.0).
[21:33:24] [DEBUG] Loading assembly NativeUI.dll ...
[21:33:24] [INFO] Found 1 script(s) in NativeUI.dll resolved to API version 2.11.6 (script assembly version: 1.7.0.0, target API version: 2.9.2).
[21:33:24] [DEBUG] Loading assembly policeimprovement.dll ...
[21:33:24] [INFO] Found 1 script(s) in policeimprovement.dll resolved to API version 2.11.6 (script assembly version: 1.0.0.0, target API version: 2.11.3).
[21:33:24] [DEBUG] Loading assembly PullMeOver.dll ...
[21:33:24] [INFO] Found 2 script(s) in PullMeOver.dll resolved to API version 2.11.6 (script assembly version: 1.0.0.0, target API version: 2.10.6).
[21:33:24] [DEBUG] Loading assembly RealLifeMod.dll ...
[21:33:24] [WARNING] Resolving API version 0.0.0 referenced in RealLifeMod.dll.
[21:33:24] [INFO] Found 1 script(s) in RealLifeMod.dll resolved to API version 2.11.6 (script assembly version: 1.0.2.0, target API version: 2.8.0).
[21:33:24] [DEBUG] Loading assembly Remembrance  LPRNS.dll ...
[21:33:24] [INFO] Found no compatible scripts in Remembrance  LPRNS.dll but loaded as a library.
[21:33:24] [DEBUG] Loading assembly WarrantV.net.dll ...
[21:33:25] [INFO] Found 1 script(s) in WarrantV.net.dll resolved to API version 3.7.0 (script assembly version: 1.0.0.0, target API version: 3.7.0).
[21:33:25] [INFO] Found 6 script(s) resolved to the deprecated API version 2.x (ScriptHookVDotNet2.dll), though the script(s) are currently running. The v2 API is deprecated and no longer actively supported. It may stop being supported in the future. You could report to the authors who developed some of the scripts that are using the deprecated API, or find alternative scripts to avoid the issue. The list of script names:
[21:33:25] [INFO] ArrestWarrant.dll
[21:33:25] [INFO] Better Chases.dll
[21:33:25] [INFO] NativeUI.dll
[21:33:25] [INFO] policeimprovement.dll
[21:33:25] [INFO] PullMeOver.dll
[21:33:25] [INFO] RealLifeMod.dll
[21:33:25] [DEBUG] Instantiating script ArrestWarrant.ArrestWarrant ...
[21:33:25] [INFO] Started script ArrestWarrant.ArrestWarrant.
[21:33:25] [DEBUG] Instantiating script BetterChases ...
[21:33:27] [INFO] Started script BetterChases.
[21:33:27] [DEBUG] Instantiating script BetterChases+PedFinder ...
[21:33:27] [INFO] Started script BetterChases+PedFinder.
[21:33:27] [DEBUG] Instantiating script NativeUI.BigMessageThread ...
[21:33:27] [INFO] Started script NativeUI.BigMessageThread.
[21:33:27] [DEBUG] Instantiating script template.Main ...
[21:33:27] [INFO] Started script template.Main.
[21:33:27] [DEBUG] Instantiating script PullMeOver.PullMeOverMain ...
[21:33:27] [INFO] Started script PullMeOver.PullMeOverMain.
[21:33:27] [DEBUG] Instantiating script PullMeOver.SpeedDisplay ...
[21:33:27] [INFO] Started script PullMeOver.SpeedDisplay.
[21:33:27] [DEBUG] Instantiating script RealLifeMod.Main ...
[21:33:32] [INFO] Started script RealLifeMod.Main.
[21:33:32] [DEBUG] Instantiating script RoadRage ...
[21:33:32] [INFO] Started script RoadRage.
[21:33:32] [DEBUG] Instantiating script WarrantV.Main ...
[21:33:33] [INFO] Started script WarrantV.Main.

**Comment by nomakewan** [comment]
@islammuhammad773-max I'm sorry, is there something in this log we should be seeing? You haven't given us any context, and nothing in this log appears to be incorrect.

**Comment by gwaub** [comment]
I know this is an old discussion but it's the first result that comes up on search engines so it might still help others.

I actually managed to figure out why for some users ScriptHookVDotNet fails to load scripts with the error "The specified user does not have a valid profile" on Proton.

Apparently .NET 4.8 checks for HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-21-xxxxxxxxxx-xxxxxxxxxx-xxxx-xxxx but it fails because that registry key is missing.

That number should be the current user's SID and needs to correspond to an actual user present in HKEY_USERS (the digits denoted by an "x" above change depending on the system).

Wine gives the current user this (bogus?) SID every time: S-1-5-21-0-0-0-1000. Proton gives the current user a valid (and unique) one instead but forgets to add the corresponding registry key listed above... until it runs wineboot at least.

Simply running "wineboot.exe -u" inside of the game's prefix creates the registry key and fixes the problem.

Proton doesn't run wineboot on prefix creation but it does so on a prefix upgrade and that happens when you manually change the Proton version used by the game and select a more recent one. This explains why it works for some users: they most likely upgraded Proton versions or somehow ran system Wine inside of the game's prefix.

TL;DR: If you have the game on Steam, you can just run protontricks, select the GTAV profile, choose "Select the default wineprefix", "Run a Wine cmd shell", type "wineboot.exe -u" (without quotes) and hit Enter. If you get a rundll32.exe error, you can safely ignore it.

**Reply by PurpleTonberry** [reply]
It was _just_ this simple, thank you so much for posting this!

**Comment by kagikn** [comment]
It's a very very late comment and it's not really for Linux enviroments, but I found a way to have SHVDN run on my Steam version copy on macOS with CrossOver, which uses Wine. Install Steam and .NET Framework 4.8 via CrossOver (and of course GTA V Legacy), install SHV and SHVDN into the game folder, add `dinput8` as a dll override on Wine.

I purchased a M5 MacBook Air for different purposes, but I could do some casual tests with SHVDN on my MBA now. It can thermal throttle so much that `powermetrics -s thermal` says `Current pressure level: Heavy` after running the game like 5 mins (with mid-high graphic settings), though. On the other hand, I haven't managed to run SHVDN on a Linux environment yet, and therefore I'm not sure how ones can make sure they have SHVDN installed into their games and run it on Linux. <img width="1710" height="1107" alt="GTAV macOS with CrossOver" src="https://github.com/user-attachments/assets/294ec8d5-2ec7-49cb-a245-1d236f8a5ecb" />

**Comment by PolygonError** [comment]
for anyone having trouble with this I documented how to get it working over here: https://github.com/Chiheb-Bacha/ScriptHookVDotNetEnhanced/discussions/17 for the enhanced version, I imagine it is exactly the same for og version.
